public/app/plugins/datasource/loki/components/LokiLabelBrowser.tsx

```diff
@@ -0,0 +1,526 @@
+import React, { Component, createRef, ChangeEvent } from 'react';
+import {
+  Button,
+  HorizontalGroup,
+  Input,
+  Label,
+  LoadingPlaceholder,
+  Popover,
+  PopoverController,
+  stylesFactory,
+  withTheme,
+} from '@grafana/ui';
+import LokiLanguageProvider from '../language_provider';
+import { css } from 'emotion';
+import store from 'app/core/store';
+import { FixedSizeList } from 'react-window';
+
+import { GrafanaTheme } from '@grafana/data';
+import { LokiLabel } from './LokiLabel';
+
+// Hard limit on labels to render
+const MAX_LABEL_COUNT = 100;
```

Comment:
Is there a reason for hard limit? Also, isn't the limit too low (I checked and our loki-ops has around ~70-80 labels, but maybe some of our users have even more)? 

**Blocker:**
If we are slicing, I think that we should let our users know that we have reached the limit and not all labels are visible. 

**Maybe next iteration/during the beta:**
We could maybe add a search to specify label names - so we wouldn't slice the labels, but just show first ~100 and then users could specify more which labels are they interested.